htlcswitch/link_test.go

```diff
@@ -1944,6 +1944,17 @@ func newSingleLinkTestHarness(chanAmt, chanReserve btcutil.Amount) (
 		return nil, nil, nil, nil, nil, nil, err
 	}
 
+	notifyUpdateChan := make(chan *contractcourt.ContractUpdate)
```

Comment:
Looks like this pattern is repeated multiple times, not sure what's the best way to deal with it. Maybe using `testify/mock` can solve it. Non-blocking tho, just saying.

Reply:
I agree, it needs a two-way ack but not sure how mock would help. The link_test & switch_test files could use a lot of work, though I didn't make it any better with this

Reply:
I think type params can help out here eventually, which will be available 🔜